pkg/driver/kubernetes.go

```diff
@@ -0,0 +1,372 @@
+package driver
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/deislabs/cnab-go/driver"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	batchclientv1 "k8s.io/client-go/kubernetes/typed/batch/v1"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	k8sContainerName    = "invocation"
+	k8sFileSecretVolume = "files"
+)
+
+// KubernetesDriver runs an invocation image in a Kubernetes cluster.
+type KubernetesDriver struct {
+	Namespace             string
+	ServiceAccountName    string
+	LimitCPU              resource.Quantity
+	LimitMemory           resource.Quantity
+	ActiveDeadlineSeconds int64
+	BackoffLimit          int32
+	SkipCleanup           bool
+	skipJobStatusCheck    bool
+	jobs                  batchclientv1.JobInterface
+	secrets               coreclientv1.SecretInterface
+	pods                  coreclientv1.PodInterface
+	deletionPolicy        metav1.DeletionPropagation
+	requiredCompletions   int32
+}
+
+// NewKubernetesDriver initializes a Kubernetes driver.
+func NewKubernetesDriver(namespace, serviceAccount string, conf *rest.Config) (*KubernetesDriver, error) {
+	driver := &KubernetesDriver{
+		Namespace:          namespace,
+		ServiceAccountName: serviceAccount,
+	}
+	driver.setDefaults()
+	err := driver.setClient(conf)
+	return driver, err
+}
+
+// Handles receives an ImageType* and answers whether this driver supports that type.
+func (k *KubernetesDriver) Handles(imagetype string) bool {
+	return imagetype == driver.ImageTypeDocker || imagetype == driver.ImageTypeOCI
+}
+
+// Config returns the Kubernetes driver configuration options.
+func (k *KubernetesDriver) Config() map[string]string {
+	return map[string]string{
+		"KUBE_NAMESPACE":  "Kubernetes namespace in which to run the invocation image",
+		"SERVICE_ACCOUNT": "Kubernetes service account to be mounted by the invocation image (if empty, no service account token will be mounted)",
+		"KUBE_CONFIG":     "Absolute path to the kubeconfig file",
+		"MASTER_URL":      "Kubernetes master endpoint",
+	}
+}
+
+// SetConfig sets Kubernetes driver configuration.
+func (k *KubernetesDriver) SetConfig(settings map[string]string) {
+	k.setDefaults()
+	k.Namespace = settings["KUBE_NAMESPACE"]
+	k.ServiceAccountName = settings["SERVICE_ACCOUNT"]
+
+	var kubeconfig string
+	if kpath := settings["KUBE_CONFIG"]; kpath != "" {
+		kubeconfig = kpath
+	} else if home := homeDir(); home != "" {
+		kubeconfig = filepath.Join(home, ".kube", "config")
+	}
+
+	conf, err := clientcmd.BuildConfigFromFlags(settings["MASTER_URL"], kubeconfig)
+	if err != nil {
+		panic(err)
```

Comment:
`KEEP CALM AND DON'T PANIC` 😺 
Wouldn't it be better to modify the `driver.Configurable` interface so `SetConfig` can return an error?

Reply:
I actually did exactly this and then realized the interface is defined in cnab-go. I'm definitely in favor of this, just has to be a separate pr.

Reply:
Yeah, this can be addressed in a follow-on PR when we can fix the interface.

Reply:
Sure I'm fine with a follow-up 👍